apps/src/applab/PlaySpaceHeader.jsx

```diff
@@ -35,12 +35,22 @@ var PlaySpaceHeader = React.createClass({
 
   render: function () {
     var leftSide, rightSide;
+    var showDataModeButton = Applab.useFirebase;
+    var toggleGroupWidth = showDataModeButton ? '160px' : '120px';
 
     if (!this.shouldHideToggle()) {
+      var toggleButtons = [
+        <button id='codeModeButton' value={ApplabInterfaceMode.CODE}>{msg.codeMode()}</button>,
+        <button id='designModeButton' value={ApplabInterfaceMode.DESIGN}>{msg.designMode()}</button>
+      ];
+      if (showDataModeButton) {
+        toggleButtons.push(
+          <button id='dataModeButton' value={ApplabInterfaceMode.DATA}>{msg.dataMode()}</button>
+        );
+      }
       leftSide = (
         <ToggleGroup selected={this.props.interfaceMode} onChange={this.props.onInterfaceModeChange}>
-          <button id='codeModeButton' value={ApplabInterfaceMode.CODE}>{msg.codeMode()}</button>
-          <button id='designModeButton' value={ApplabInterfaceMode.DESIGN}>{msg.designMode()}</button>
+          {toggleButtons}
```

Comment:
Instead of putting these all in a toggleButtons variable, I think you have the following
```
<ToggleGroup selected={this.props.interfaceMode} onChange={this.props.onInterfaceModeChange}>
  <button id='codeModeButton' value={ApplabInterfaceMode.CODE}>{msg.codeMode()}</button>
  <button id='designModeButton' value={ApplabInterfaceMode.DESIGN}>{msg.designMode()}</button>
  {showDataModeButton && <button id='dataModeButton' value={ApplabInterfaceMode.DATA}>{msg.dataMode()}}</button>
</ToggleGroup>
```
I find components end up being easier to understand when as much as the DOM structure is together as possible.